### PR TITLE
Fix incomplete device path on OSX

### PR DESCRIPTION
**Pull request by klmp200**
I assumed the filtered content contained the /dev/ path but that wasn't the case.

This is now fixed

### CODE REVIEW

**Comment by puzrin** [comment]
Released. Did you tested that on real device?

**Comment by klmp200** [comment]
Yes, I'm currently printing with it on my macbook :)
Last time I quickly re-aranged my hardcoded workaround at 2AM and only checked the scan part :/
I've been printing with that version for 3 hours now :)

**Comment by puzrin** [comment]
Any connectivity issues? As far as I remember, PT-P300BT was freezing after a few minutes in Linux and I had to off/on the power to reconnect. May be, due bug in `.readSync()`. I switched to DYMO PNP, it's nice but more difficult to purchase (PT-P300BT available via Aliexpress worldwide).

**Comment by klmp200** [comment]
I'm sometimes having some issues where the printer goes in error mode and I have to press the power button
Also, sometimes it's not responding at all after a reboot, when that happens, I unpair/pair the device again and it goes again.

The readSync bug was something else all together, I had node printing out the following stacktrace :

```bash
➜  lbprint git:(master) ✗ ./lbprint.js 'foo bar' baz
TypeError [ERR_INVALID_ARG_TYPE]: The "options" argument must be of type object. Received type number (32)
    at Object.readSync (node:fs:733:7)
    at get_status (/Users/<user>/lbprint/lib/drivers/brother_ptp300bt.js:117:8)
    at module.exports.print (/Users/<user>/lbprint/lib/drivers/brother_ptp300bt.js:190:3)
    at module.exports.run (/Users/<user>/lbprint/lib/cli.js:169:31)
```

**Comment by puzrin** [comment]
Hm... then i don't know why it worked at all with old `.readSync()` call, in linux. There are no records in docs about such API change (looks like call was initially bad).

If you find a way to improve connectivity - let me know. I had not enough patience to solve that.

**Comment by gmdelisle** [comment]
Hi @klmp200 and others.

MacOS user here also, not a developer though :(
Wondering how you got this to work in MacOS.

This does not work in MacOS to connect to the printer via Bluetooth.
```
# Get MAC address of PT-P300BT**** (remember it)
hcitool scan
# Connect to printer
sudo rfcomm connect 0 <MAC_ADDRESS> 1 &
```
So I connect to the printer using [BluetoothConnector](https://github.com/lapfelix/BluetoothConnector) (installed via Brew)

Which seems to work. 
SCR-20240602-ulys

Then installed **lbprint** with this command from the Readme.

```
# local install from repo
git clone https://github.com/puzrin/lbprint.git
cd lbprint
npm install
```

When I try printing I get this error.
SCR-20240602-ukxv

Printer seems to instantly disconnect.
Any help/tutorial on your steps to get it working in MacOS welcome.
Thank you! 🙏

**Comment by klmp200** [comment]
Have you tried adding the device with the macos bluetooth manager ?
That's what I did during my testing and it works great on my side.

Very interesting feedback, thanks :)

**Comment by gmdelisle** [comment]
Yeah I did and unfortunately, it connects then instantly disconnects.
Feels like it recognizes it's not connected to a mobile phone, so it kills the connection since it connects fine on mobile. Dont get how you guys got this to work on Linux/MacOS. 
I thought you maybe used a script that makes the Bluetooth connection stick. 🤷‍♂️
Thanks for the reply.

**Comment by gmdelisle** [comment]
Dod you have to change any permissions via Terminal for lbprint.js to be able to access connected bluetooth devices?

**Comment by klmp200** [comment]
I haven't done anything special.
I've upgraded to Sonoma recently, I'll check if that changes something.

**Comment by klmp200** [comment]
Well, I tested on Sonoma and it's still working fine, I can't reproduce your issue, I'm sorry :/

**Comment by gmdelisle** [comment]
Thank you so much for trying! 🙏

Hope the owner of this repo adds Mac instructions at some point si Noobs like me can figure it out.

**Comment by klmp200** [comment]
> Yeah I did and unfortunately, it connects then instantly disconnects. Feels like it recognizes it's not connected to a mobile phone, so it kills the connection since it connects fine on mobile. Dont get how you guys got this to work on Linux/MacOS. I thought you maybe used a script that makes the Bluetooth connection stick. 🤷‍♂️ Thanks for the reply.

I'm thinking about it again.
When I do connect through the bluetooth manager in macos, it doesn't stay connected and that's expected.
It creates a virtual device in /dev and that's what the tool is then using.
When the tool talks to the file, it does init the connection itself, you don't need to manually connect each time.

TLDR is: have you tried to use the tool after pairing the device normally with macos ? Even though the connection doesn't stay open ?

**Comment by gmdelisle** [comment]
Hi!
I have connected to the printer in two ways :
- Through MacOS settings
- Through BluetoothConnector in terminal (installed via Brew)
image

In both cases it does show an initial working connection.
Indeed I have tried using the tool (lbprint) immediately after this connection.

If however I needed to do anything else to either :
- Create a virtual device in /dev
- init the connection

I have not done & don't know how to do.

Thanks for the response!
